libparsec/crates/client/src/workspace/store/resolve_path.rs

```diff
@@ -89,7 +90,10 @@ pub(super) async fn resolve_path(
 ) -> Result<(ArcLocalChildManifest, PathConfinementPoint), ResolvePathError> {
     resolve_path_maybe_lock_for_update(store, path, false)
         .await
-        .map(|(manifest, confinement_point, _)| (manifest, confinement_point))
+        .map(|(manifest, confinement_point, lock_guard)| {
+            debug_assert!(lock_guard.is_none());
```

Comment:
Is that a debugging artifact?

Reply:
nope, it's a proper guard

`debug_assert` means the check is only present in debug build, which is because this assert is very defensive (since there is a strong and direct relation between the `false` passed to `resolve_path_maybe_lock_for_update` and the `None` returned).